lib/Sema/CSBindings.cpp

```diff
@@ -1193,12 +1170,12 @@ std::optional<BindingSet> ConstraintSystem::determineBestBindings(
     // associated with given type variable, any default constraints,
     // or any conformance requirements to literal protocols with can
     // produce a default type.
-    bool isViable = isViableForRanking(bindings);
+    bool isViable = bindings.hasViableBindings() || bindings.isDirectHole();
```

Comment:
This is a functional change `isViableForRanking` had more logic than just `bool(bindings)`.

Reply:
We discussed this offline and it appears that everything should behave the same way as before.